package/gluon-mesh-babel/src/babelparse.h

```diff
@@ -0,0 +1,67 @@
+/*
+   Copyright (c) 2017, Christof Schulze <christof.schulze@gmx.net>
+   All rights reserved.
+
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are met:
+
+   1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+   2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+   FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+   SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+   CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+   OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+   */
+
+#pragma once
+
+#include <stdio.h>
+#include <string.h>
+#include <netdb.h>
+#include "errno.h"
+#include <arpa/inet.h>
+#include <stdlib.h>
+#include <json-c/json.h>
+#include <unistd.h>
+
+#define BABEL_PORT 33123
+#define SOCKET_INPUT_BUFFER_SIZE 255
+
+struct babelneighbour {
+	char *action;
+	char *address_str;
+	char *ifname;
+	int reach;
+	int rxcost;
+	int txcost;
+	int cost;
+};
+
+struct babelroute{
+	char *action;
+	char *route;
+	char *prefix;
+	char *from;
+	char *id;
+	int metric;
+	int refmetric;
+	char *via;
+	char *ifname;
+	struct in6_addr in6_via;
+};
+
+struct babelneighbour babelparse_get_neighbour(char *line);
+struct babelroute babelparse_get_route(char *line);
+
+// static void input_pump(int fd, struct json_object* obj, void (*lineprocessor)(char*, struct json_object*));
+void babelparse_readbabeldata(struct json_object *obj, void (*lineprocessor)(char*, struct json_object*));
```

Comment:
Are there any specific reasons, why you decided to pass a `json_object*` instead of a generic `void*`? This would make it possible to use the library without linking to json-c. But I'm not sure whether this will happen some day. So maybe it's irrelevant.